dotnet-statsig/src/Statsig/Server/Evaluation/Evaluator.cs

```diff
@@ -64,6 +64,178 @@ internal ConfigEvaluation GetLayer(StatsigUser user, string layerName)
             return Evaluate(user, _store.LayerConfigs[layerName]);
         }
 
+        internal Dictionary<string, Object> GetAllEvaluations(StatsigUser user)
+        {
+            if (!_initialized)
+            {
+                return null;
+            }
+
+            var gates = new Dictionary<string, Dictionary<string, object>>();
+            foreach (var kv in _store.FeatureGates)
+            {
+                if (kv.Value.Entity == "segment") 
+                {
+                    continue;
+                }
+
+                var hashedName = HashName(kv.Value.Name);
+                var gate = Evaluate(user, kv.Value).GateValue;
+                var entry = new Dictionary<string, object>
+                {
+                    ["name"] = hashedName,
+                    ["value"] = gate.Value,
+                    ["rule_id"] = gate.RuleID,
+                };
+                if (gate.SecondaryExposures.Count > 0)
+                {
+                    entry["secondary_exposures"] = CleanExposures(gate.SecondaryExposures);
+                }
+                gates.Add(hashedName, entry);
+            }
+            
+            var dynamicConfigs = new Dictionary<string, Dictionary<string, object>>();
+            foreach (var kv in _store.DynamicConfigs)
+            {
+                var hashedName = HashName(kv.Value.Name);
+                var config = Evaluate(user, kv.Value).ConfigValue;
+                var entry = ConfigSpecToInitResponse(hashedName, kv.Value, config);
+                if (kv.Value.Entity != "dynamic_config")
+                {
+                    entry["is_experiment_active"] = IsExperimentActive(kv.Value);
+                    entry["is_user_in_experiment"] = IsUserAllocatedToExperiment(user, kv.Value);
+                }
+                dynamicConfigs.Add(hashedName, entry);
+            }
+
+            var layerConfigs = new Dictionary<string, Dictionary<string, object>>();
+            foreach (var kv in _store.LayerConfigs)
+            {
+                var hashedName = HashName(kv.Value.Name);
+                var evaluation = Evaluate(user, kv.Value);
+                var config = evaluation.ConfigValue;
+                var entry = ConfigSpecToInitResponse(hashedName, kv.Value, config);
+
+                if (!string.IsNullOrWhiteSpace(evaluation.ConfigDelegate))
+                {
+                    entry["allocated_experiment_name"] = HashName(evaluation.ConfigDelegate);
+                    entry["is_experiment_active"] = true;
+                    entry["is_user_in_experiment"] = true;
```

Comment:
hmm this was "good enough" for node because sticky layers didnt exist at the time, but since this is actually replacing `initialize` for everyone, I think it would break for people using stick layer experiments on mobile

Reply:
What's the right way to evaluate this?

Reply:
is_experiment_active is something we probably need to pass down from download_config_specs

is_user_in_experiment will need us to find a specific rule of that experiment and evaluate to see if the user fails that rule. Let's sync offline

Reply:
Oh looks like you are already doing this for experiments. I think for layers it's just a matter of doing the same for the "allocated experiment"